### PR TITLE
webui: Add polling to `SearchResultsLoadSensor`.

### PR DESCRIPTION
**Pull request by junhaoliao**
# References
Internally, it was reported that the `SearchResultsLoadSensor` does not load new results and keeps displaying a loading status. Further investigation discovered that when the browser client height exceeds the height of newly loaded results, the newly loaded results do not overflow enough to make the `SearchResultsLoadSensor` go below the bottom of visible browser client window. As a result, the `IntersectionObserver` callback is not called to load more results, causing a visually "hanged" `SearchResultsLoadSensor`. e.g.,
image

# Description
1. Add polling to `SearchResultsLoadSensor`.

# Validation performed
1. Without the changes, replicated the issue as reported in "References" with a browser window of odd height (> 2000 px).
2. Applied the changes and observed the `SearchResultsLoadSensor` go below the bottom of the browser window after two loads.

### CODE REVIEW

File: components/webui/imports/ui/SearchView/SearchResultsTable.jsx [hunk 1]
```diff
@@ -26,18 +31,20 @@ const SearchResultsLoadSensor = ({
             return;
         }
 
-        const observer = new IntersectionObserver(
-            (entries) => {
+        let observer = null;
```
Comment (kirkrodrigues) [inline components/webui/imports/ui/SearchView/SearchResultsTable.jsx:34]: I feel it's more intuitive to write something like:

```js
    useEffect(() => {
        if (false === hasMoreResults) {
            return;
        }

        let observer = new IntersectionObserver((entries) => {
            if (entries[0].isIntersecting) {
                loadInterval.current = setInterval(() => {
                    onLoadMoreResults();
                }, SEARCH_RESULTS_LOAD_SENSOR_POLL_INTERVAL_MS);
            } else {
                if (null !== loadInterval.current) {
                    clearInterval(loadInterval.current);
                    loadInterval.current = null;
                }
            }
        });
        observer.observe(loadingBlockRef.current);

        return () => {
            observer.disconnect();
            if (null !== loadInterval.current) {
                clearInterval(loadInterval.current);
            }
        };
    }, [hasMoreResults]);
```

The idea being while the loader is in view there are more results, we periodically try to load more results and when the loader goes out of view, we disable the periodic task. What do you think?

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: components/webui/imports/ui/SearchView/SearchResultsTable.jsx (hunk 1 above)
Comment (kirkrodrigues) [reply components/webui/imports/ui/SearchView/SearchResultsTable.jsx:34]: Also, I don't think we need the `hasMoreResults` check in the IntersectionObserver handler, right?

Comment (junhaoliao) [reply components/webui/imports/ui/SearchView/SearchResultsTable.jsx]: > Also, I don't think we need the hasMoreResults check in the IntersectionObserver handler, right?

Right, the loading sensor style `visibility` changes to `hidden` when `hasMoreResults === false` and therefore `entries[0].isIntersecting` does not evaluate to `true` in all conditions.

Comment (junhaoliao) [reply components/webui/imports/ui/SearchView/SearchResultsTable.jsx]: > The idea being while the loader is in view and there are more results, we periodically try to load more results and when the loader goes out of view, we disable the periodic task. What do you think?

I agree. The proposed approach is cleaner and avoid unnecessary polling.

**Review by kirkrodrigues — APPROVED** [review]
How about:

webui: Poll for results while `SearchResultsLoadSensor` is visible. (#315)

**Pull request merged**